components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java

```diff
@@ -257,35 +254,60 @@ public static TicketGrantingTicket getTicketGrantingTicket(String ticketGranting
 
             // Get all supported claims
             ClaimManager claimManager = userRealm.getClaimManager();
-            org.wso2.carbon.user.api.ClaimMapping[] mappings = claimManager.getAllSupportClaimMappingsByDefault();
-
+            org.wso2.carbon.user.api.ClaimMapping[] mappings = claimManager.getAllClaimMappings();
             for (org.wso2.carbon.user.api.ClaimMapping claimMapping : mappings) {
                 requestedClaims.add(claimMapping.getClaim().getClaimUri());
+                log.debug("adding requested claim: " + claimMapping.getClaim().getClaimUri());
             }
 
             // Get claim values for the user
-            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            UserStoreManager userStoreManager = null;
+            boolean localAuthentication = false;
+            try {
+                userStoreManager = userRealm.getUserStoreManager();
+                localAuthentication = userStoreManager.isExistingUser(username);
+            }
+            catch (UserStoreException e) {
+                // User came from federated authentciation
+                log.warn ("Error while retrieving the user from federated authentication, e");
```

Comment:
I have changed the warn log to error  and added the UserStore exception to increase the severity of the error. Please find the fixed in following commit
7042a9def49a60d0d96f05a107e7d771566369d6